ChildProject/pipelines/eafbuilder.py

```diff
@@ -1,28 +1,120 @@
+        :param template: name of the template to use (basic, native, or non-native)
+        :type template: str
+        :param context_onset: context onset and segment offset difference in milliseconds, 0 for no introductory context
+        :type context_onset: float
+        :param context_offset: context offset and segment offset difference in milliseconds, 0 for no outro context
+        :type context_offset: float
+        """
+
+        # TODO: Make sure etf file paths are approprite and robust. 
+        etf_path = "{}.etf".format(template)
+        psfx_path = "{}.pfsx".format(template)
+
+        print("making the "+eaf_type+" eaf file and csv")
+
+        segments = pd.read_csv(segments)
+
+        for recording_filename, segs in segments.groupby('recording_filename'):
+            recording_filename = os.path.splitext(recording_filename)[0]
+            output_filename = recording_filename + '_' + eaf_type + '_' + template
+            
+            # TODO: This list of timestamps as tuples might not be ideal/should perhaps be optimized, but I am just replicating the original eaf creation code here.
+            timestamps = [(on, off) for on, off in segs.loc[:, ['segment_onset', 'segment_offset']].values]
+
+            output_dir = os.path.join(destination, recording_filename)
+            with resources.path('ChildProject.templates', etf_path) as e_path:
+                create_eaf(
+                    e_path,
+                    output_filename,
+                    output_dir,
+                    timestamps,
+                    eaf_type,
+                    context_onset,
+                    context_offset,
+                    template
+                )
+
+            with resources.path('ChildProject.templates', psfx_path) as p_path:
+                shutil.copy(p_path, os.path.join(output_dir, "{}.pfsx".format(output_filename)))
 
-        pass
 
     @staticmethod
     def setup_parser(parser):
-        parser.add_argument("path", help = "project path")
-        parser.add_argument("destination", help = "eaf destination")
+        parser.add_argument("--destination", help = "eaf destination")
         parser.add_argument('--segments', help = 'path to the input segments dataframe', required = True)
-        parser.add_argument('--type', help = 'eaf-type', choices = ['random', 'periodic'])
-        parser.add_argument('--template', help = 'eaf template', choices = ['basic', 'native', 'non-native'])
-        parser.add_argument('--context-onset', help = 'context onset and segment offset difference, 0 for no introductory context', type = float, default = 0)
-        parser.add_argument('--context-offset', help = 'context offset and segment offset difference, 0 for no outro context', type = float, default = 0)
+        # TODO: add other options here such as high-volubility, energy, etc.?
+        parser.add_argument('--eaf-type', help = 'eaf-type', choices = ['random', 'periodic'], required = True)
```

Comment:
agreed! not totally sure about the energy one being used a lot in the future, but high vol is very common (as you saw in yesterday's talk)

Reply:
so maybe we should allow arbitrary values for this parameter ?